src/schema/v2/viewingRoom.ts

```diff
@@ -0,0 +1,312 @@
+          }
+
+          if (endAt < now) {
+            return null
+          }
+
+          if (short) {
+            if (endAt > now.clone().add(5, "days")) {
+              return null
+            }
+          } else {
+            if (endAt > now.clone().add(30, "days")) {
+              return null
+            }
+          }
+
+          return `${moment
+            .duration(endAt.diff(now))
+            .locale(LocaleEnViewingRoomRelativeLong)
+            .humanize(false, { ss: 1, d: 31 })}`
+        },
+      },
+      distanceToOpen: {
+        type: GraphQLString,
+        args: {
+          short: {
+            type: new GraphQLNonNull(GraphQLBoolean),
+            defaultValue: false,
+          },
+        },
+        resolve: ({ start_at: _startAt }, { short = false }) => {
+          if (_startAt === null) {
+            return null
+          }
+
+          const startAt = moment(_startAt)
+          const now = moment()
+
+          if (startAt < now) {
+            return null
+          }
+
+          if (!short && startAt > now.clone().add(30, "days")) {
+            return null
+          }
+
+          const distance = moment.duration(startAt.diff(now))
+          return distance
+            .locale(
+              short
+                ? LocaleEnViewingRoomRelativeShort
+                : LocaleEnViewingRoomRelativeLong
+            )
+            .humanize(short, { ss: 1, d: 31 })
+        },
+      },
+      endAt: {
+        // TODO: type should be ISO8601DateTime
+        type: GraphQLString,
+        description:
+          "Datetime after which the viewing room is no longer viewable",
+        resolve: ({ end_at }) => end_at,
+      },
+      exhibitionPeriod: {
+        type: GraphQLString,
+        resolve: ({ start_at, end_at }) => dateRange(start_at, end_at, "UTC"),
+      },
+      firstLiveAt: {
+        type: GraphQLString,
+        description: "Datetime when viewing room first viewable",
+        resolve: ({ first_live_at }) => first_live_at,
+      },
+      heroImageURL: {
+        type: GraphQLString,
+        deprecationReason: "Use image field instead",
+        resolve: ({ hero_image_url }) => hero_image_url,
+      },
+      href: {
+        type: GraphQLString,
+        resolve: ({ slug }) => `/viewing-room/${slug}`,
+      },
+      // TODO: should be ARImageType
+      image: {
+        type: GravityARImageType,
+        resolve: ({ image }) => {
+          return image
+        },
+      },
+      introStatement: {
+        type: GraphQLString,
+        description: "Introductory paragraph",
+        resolve: ({ intro_statement }) => intro_statement,
+      },
+      partner: {
+        type: PartnerType,
+        resolve: ({ partner_id }, _args, { partnerLoader }) => {
+          return partnerLoader(partner_id)
+        },
+      },
+      // TODO: it causes yarn dump:local to fail with typeError: Cannot read properties of undefined (reading 'connectionType') in partnerArtworks.ts
```

Comment:
I could use some help with this. I’ve spent too much time on it. I’ll try again with a clear head, but I’m not sure if I will manage

Reply:
Might need to use `require` on the type instead of import like in https://github.com/artsy/metaphysics/blob/aea27c604710c6dc935ca8f5f8cffd73a557c2a8/src/schema/v2/artist/highlights.ts#L17

Reply:
@mzikherman yes your suggestions works! Thanks a lot!